src/node/internal/internal_zlib.ts

```diff
@@ -115,19 +153,30 @@ function normalizeArgs(
   throw new ERR_INVALID_ARG_TYPE('callback', 'Function', callbackOrUndefined);
 }
 
-function wrapCallback(callback: CompressCallback): InternalCompressCallback {
-  return (res: Error | ArrayBuffer) => {
+function processChunkCaptureError(
+  engine: ZlibBase,
+  data: ArrayBufferView | string,
+  cb: CompressCallback
+): void {
+  try {
+    const res = processChunk(engine, data);
     queueMicrotask(() => {
-      if (res instanceof Error) {
-        callback(res);
-      } else {
-        callback(null, Buffer.from(res));
-      }
+      cb(null, res);
     });
-  };
+  } catch (err: unknown) {
+    if (err instanceof Error) {
+      queueMicrotask(() => {
+        cb(err);
+      });
+      return;
+    }
+
+    throw err;
```

Comment:
Technically this is possible, but it will only be possible if we have a bug on our implementation right? In order to find the bugs easier, what about using `.cause`?

```suggestion
    const unreachable = new Error('Unreachable')
    unreachable.cause = err;
    throw unreachable;
```